Source/WebKit/UIProcess/API/wpe/WebKitWebViewWPE.cpp

```diff
@@ -326,3 +326,22 @@ void webkit_web_view_toggle_inspector(WebKitWebView* webView)
         inspector->show();
 }
 #endif
+
+/**
+ * webkit_web_view_get_theme_color:
+ * @web_view: a #WebKitWebView
+ * @color: (out): a #WebKitColor to fill in with the theme color
+ *
+ * Gets the theme color that is specified by the content in the @web_view.
+ *
+ * Since: 2.50
+ */
+void
+webkit_web_view_get_theme_color(WebKitWebView* webView, WebKitColor* color)
+{
+    g_return_if_fail(WEBKIT_IS_WEB_VIEW(webView));
+    auto& page = webkitWebViewGetPage(webView);
+
+    g_return_if_fail(page.themeColor().isValid());
```

Comment:
Checking the implementation of `Page::themeColor()` and `Document::themeColor()` I see that there are two cases in which the color may end up being invalid:

- There is no top-level `Document` (for example: may happen during loads, temporarily)
- The theme color cannot be determined (not declared with the corresponding `<meta>` element, no app-manifest, etc.)

These are cases that we should handle. The `Color` class has a comment telling that it can represent a *Special "invalid color" state, treated as transparent black but distinguishable*.

Taking the above into account, I think this function should return `gboolean` to indicate whether the color is valid. In case of getting an invalid color, fill the `WebKitColor`, with `red = green = blue = alpha = 0.0`.